### PR TITLE
Extend frsky smartport telemetry for flightmodes, satcount and fixstate

### PR DESCRIPTION
**Pull request by thedevleon**
This continues the discussion from https://github.com/PX4/Firmware/pull/3762.
I remapped the flight mode numbers sent in accordance to my suggested change to the script here https://github.com/ilihack/LuaPilot_Taranis_Telemetry/pull/2 - to make the script work flawlessly with APM as well as PX4.

### CODE REVIEW

File: src/drivers/frsky_telemetry/sPort_data.c [hunk 1]
```diff
@@ -234,41 +234,61 @@ void sPort_send_T1(int uart)
 	memset(&vehicle_status, 0, sizeof(vehicle_status));
 	orb_copy(ORB_ID(vehicle_status), vehicle_status_sub, &vehicle_status);
 
-	/* map flightmode from PX4 to APM */
-	uint32_t flightmodePX4 = (int)(vehicle_status.main_state);
-	uint32_t flightmodeAPM = 0;
+	/* Map Flightmodes from 18-29 */
+	uint32_t flightmode = (int)(vehicle_status.main_state);
```
Comment (kd0aij) [inline src/drivers/frsky_telemetry/sPort_data.c]: does this compile for you?

Comment (thedevleon) [inline src/drivers/frsky_telemetry/sPort_data.c]: Just trying to compile now... Will report back as soon as I get it compiled.

**Comment by thedevleon** [comment]
Corrected some spelling mistakes, now everything compiles nicely :)

**Comment by thedevleon** [comment]
Closing this for now since I want to make a couple more changes on a freshly forked repo -  this was based on a very outdated fork - and rebase is too much work.

**Pull request closed without merging**